src/test/java/com/productdock/library/rental/domain/BookRentalRecordShould.java

```diff
@@ -20,7 +20,7 @@ class BookRentalRecordShould {
     private UserBookActivity userBookActivity;
 
     @Test
-    void addRentWhenUserAlreadyReservedTheBook() {
+    void addRentRecord_whenUserAlreadyReservedTheBook() {
```

Comment:
In `addRentRecord_whenUserAlreadyReservedTheBook` test, and `addRentRecord_whenUserHadNotRentedOrReservedItAlready` test, 
It is possible to replace all 3 assert lines with one assertion:
`assertThat(bookRentalRecord.getBookCopies()).containsOnly(rentBookCopy);`

This code is not responsible for actually creating a new BookCopy, so we don't need to verify patron and new status. We just want to know that this code knows how to manipulates the bookCopies list.